plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -529,64 +535,75 @@ func (r *ReferenceManager) ensureSeedReferences(ctx context.Context, seed *core.
 	return r.lookupSecret(ctx, seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name)
 }
 
-func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes admission.Attributes, shoot *core.Shoot) error {
-	if _, err := r.cloudProfileLister.Get(shoot.Spec.CloudProfileName); err != nil {
-		return err
-	}
+func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes admission.Attributes, oldShoot, shoot *core.Shoot) error {
+	shootCreation := oldShoot == nil
```

Comment:
In the ShootValidator for example, we initialize the `oldShoot` with an empty `&gardencorev1beta1.Shoot{}` structure so that we don't have to have such `shootCreation` variable. WDYT about doing the same here?

Also, what about the other resources in this admission plugin? I haven't checked them, but I assume we should behave similarly for them?

Reply:
As discussed, I addressed your feedback about the empty shoot assignment and will adjust checks for other resources in another PR in the future due to time restrictions.